src/main/java/calculator/CalculatorServiceImpl.java

```diff
@@ -0,0 +1,134 @@
+package calculator;
+
+import com.google.code.mathparser.MathParser;
+import com.google.code.mathparser.MathParserFactory;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import repository.Repository;
+import repository.RepositoryFactory;
+import responseCodes.codes.StatusCode;
+import responseCodes.exception.BadRequestException;
+import responseCodes.exception.ForbiddenException;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static constants.ControllerConstants.ABS_RANGE;
+import static constants.ControllerConstants.DIVIDE;
+import static constants.ControllerConstants.EMPTY_SYMBOL;
+import static constants.ControllerConstants.EXPRESSION;
+import static constants.ControllerConstants.MINUS;
+import static constants.ControllerConstants.MULTIPLY;
+import static constants.ControllerConstants.PARSING_INFO;
+import static constants.ControllerConstants.PLUS;
+import static utils.ConversionUtil.deleteSpacesAndConvertListToString;
+
+
+public class CalculatorServiceImpl implements CalculatorService {
+    private static final RepositoryFactory REPOSITORY_FACTORY = new RepositoryFactory();
+    private static final Repository REPOSITORY = REPOSITORY_FACTORY.createRepository();
+    private static final Logger logger = LogManager.getLogger(CalculatorServiceImpl.class);
+
+    @Override
+    public int calculate(String id) {
+        String expression = getFinalExpression(id);
+        MathParser mathParser = MathParserFactory.create();
+
+        return mathParser.calculate(expression)
+                .doubleValue()
+                .intValue();
+    }
+
+    @Override
+    public StatusCode addData(String id, String parameterName, String paramValue) throws Exception {
+        Optional<String> valueFromRepository = REPOSITORY.getValue(id, parameterName);
+        if (!REPOSITORY.existData(id)) {
+            REPOSITORY.putNewData(id);
```

Comment:
As user of repository, I do not want to care about inner structure of repository. I just want to add something in repository and retrieve something from it,
Move this check to repository level.